pkg/controller/build/reconciler.go

```diff
@@ -1328,6 +1360,149 @@ func (b *buildReconciler) syncMachineConfigPools(ctx context.Context) error {
 // if needed.
 func (b *buildReconciler) syncMachineConfigPool(ctx context.Context, mcp *mcfgv1.MachineConfigPool) error {
 	return b.timeObjectOperation(mcp, syncingVerb, func() error {
-		return b.createNewMachineOSBuildOrReuseExistingForPoolChange(ctx, mcp)
+		return b.reconcilePoolChange(ctx, mcp)
+	})
+}
+
+func (b *buildReconciler) reconcilePoolChange(ctx context.Context, mcp *mcfgv1.MachineConfigPool) error {
+
+	mosc, err := utils.GetMachineOSConfigForMachineConfigPool(mcp, b.utilListers())
+	if err != nil {
+		// Not opted in yet, so nothing for us to do.
+		klog.Infof("buildReconciler: no MachineOSConfig for pool %q, skipping", mcp.Name)
```

Comment:
Shouldn't we only return this if the error is NotFound? Otherwise we're masking other potential errors